### ISSUE TITLE
Default values and abstract conversion

### ISSUE THREAD

**Issue by Gama11** [elnabo/json2object#17]
I just tested #13 - seems to work nicely, except it can't handle the abstract conversion here. If a` is of type `Array<String>` it works.

```haxe
class Main {
    public static function main() {
        var parser = new json2object.JsonParser<Struct>();
    }
}

typedef Struct = {
    @:optional @:default([])
    var a:ReadonlyArray<String>;
}

abstract ReadonlyArray<T>(Array<T>) from Array<T> {}
```

![](http://i.imgur.com/ztbXSR7.png)

Since `ReadonlyArray` has a `from` defined and is thus compatible with `[]`, I think this should work.

**Comment by elnabo** [comment]
Ok this should be fixed by c92be775b0b8347d36fe4d1170cc41c1a3c08563

**Comment by Gama11** [comment]
Seems to work fine now, thanks.

Keep up the great work, json2object is starting to become a really solid lib. :)
